tests/safe_mode/check_safe_mode.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2019 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""script for checking safeMode in DML is as expected
+
+In test case we send `update` SQL before, in and after a specific sharding DDL
+procedure, and this script checks that update binlog event added during sharding
+DDL not synced phase has safeMode true, otherwise the safeMode is false.
+"""
+
+#!/usr/bin/env python
+# coding: utf-8
+
+import json
+import urllib
+from collections import defaultdict
+
+
+class TraceType(object):
+    BinlogEvent = 1
+    JobEvent = 2
+
+
+class OpType(object):
+    UPDATE = 2
+    DDL = 4
+
+
+class ReplicationEvent(object):
+    QUERY_EVENT = 2
+    UPDATE_ROWS_EVENTv0 = 21
+    UPDATE_ROWS_EVENTv1 = 24
+    UPDATE_ROWS_EVENTv2 = 31
+
+    UPDATE = [UPDATE_ROWS_EVENTv0, UPDATE_ROWS_EVENTv1, UPDATE_ROWS_EVENTv2]
+
+
+def scan_events():
+    url = 'http://127.0.0.1:8264/events/scan?limit=1000'
```

Comment:
is 1000 enough? should we not set limit argument?

Reply:
added new scan iterator in test scripts